subprojects/xsts/xsts-analysis/src/main/kotlin/hu/bme/mit/theta/xsts/analysis/XstsInUnWrappers.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright 2024 Budapest University of Technology and Economics
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package hu.bme.mit.theta.xsts.analysis
+
+import hu.bme.mit.theta.analysis.Prec
+import hu.bme.mit.theta.analysis.expr.ExprState
+import hu.bme.mit.theta.analysis.unit.UnitPrec
+import hu.bme.mit.theta.analysis.unit.UnitState
+
+fun <S : ExprState> xstsCombineStates(xstsState: XstsState<UnitState>, dataState: S): XstsState<S> {
+    return XstsState.of(dataState, xstsState.lastActionWasEnv(), xstsState.isInitialized)
+}
+
+fun xstsStripState(xstsState: XstsState<*>): XstsState<UnitState> {
+    return XstsState.of(UnitState.getInstance(), xstsState.lastActionWasEnv(), xstsState.isInitialized)
+}
+
+fun <S : ExprState> xstsExtractFromState(xstsState: XstsState<S>): S {
+    return xstsState.state
+}
+
+val xstsStripPrec: (Prec) -> Prec = { it }
+
```

Comment:
xstsStripPrec and xstsStripPrecFun are really confusing names
Maybe extractControlPrec and extractDataPrec?

Reply:
Resolved in 4f253560e834e30216e11aed4c75f48673ea4f6c